Libraries/LibWeb/Selection/Selection.cpp

```diff
@@ -134,8 +135,15 @@ String Selection::direction() const
 // https://w3c.github.io/selection-api/#dom-selection-getrangeat
 WebIDL::ExceptionOr<GC::Ptr<DOM::Range>> Selection::get_range_at(unsigned index)
 {
-    // The method must throw an IndexSizeError exception if index is not 0, or if this is empty.
-    if (index != 0 || is_empty())
+    GC::Ptr<DOM::Document> selection_document = document();
+    GC::Ptr<DOM::Node> focus = focus_node();
+    GC::Ptr<DOM::Node> anchor = anchor_node();
+
+    // The method must throw an IndexSizeError exception if index is not 0, or if this is empty or either focus or anchor is not in the document tree.
+    auto is_focus_in_document_tree = focus && &focus->document() == selection_document;
+    auto is_anchor_in_document_tree = anchor && &anchor->document() == selection_document;
```

Comment:
The `document()` on line 138 gives you a `GC::Ref`, not a `GC::Ptr`, and you can directly compare these values here.